heat/core/operations.py

```diff
@@ -123,13 +123,52 @@ def allclose(x, y, rtol = 1e-05, atol = 1e-08, equal_nan = False):
 
     """
 
-    if not isinstance(x, tensor.tensor):
-        raise TypeError('Expected x to be a ht.tensor, but was {}'.format(type(x)))
+    if isinstance(x, tensor.tensor):
+        pass
+
+    elif np.isscalar(x):
+        try:
+            x = tensor.array([float(x)])
+        except (ValueError, TypeError,):
+            raise TypeError('Data type not supported, input was {}'.format(type(x)))
+
+    else:
+        raise TypeError('Only tensors and numeric scalars are supported, but input was {}'.format(type(x)))
+
+    if isinstance(y, tensor.tensor):
+        pass
+
+    elif np.isscalar(y):
+        try:
+            y = tensor.array([float(y)])
+        except (ValueError, TypeError,):
+            raise TypeError('Data type not supported, input was {}'.format(type(y)))
+
+    else:
+        raise TypeError('Only tensors and numeric scalars are supported, but input was {}'.format(type(y)))
+
+    if (x.split is not None) and (y.split is not None) and (x.split != y.split):
+        # It is NOT possible to perform allclose on tensors with different splits
+        raise NotImplementedError('Not implemented for different splittings of tensors')
+
+    # perform local allclose
+    # no sanitization for shapes of x and y needed, torch.allclose raises relevant errors
+    _local_allclose = torch.tensor(torch.allclose(x._tensor__array, y._tensor__array, rtol, atol, equal_nan))
+
+    if x.comm.is_distributed():
+        # Makes no distinction if y is also distributed. IS THAT VALID?
```

Comment:
If y is also distributed you have to redistribute it first as to make the axes consistent.This requires to merge PR #165 first as the implementation is in there.

Reply:
Funny enough, I tried out this feature in PR #165 and it didn't work (at least locally). I'll wait for the PR to be merged before I continue

Reply:
What feature does not work?

Reply:
Has been resolved with help from daniel, the issue was rather my ignorance than the feature itself... turns out if you want to check the distribution of a tensor you _actually have to distribute it_ ^^